### ISSUE TITLE
Eratic behaviour of successive 'npm install' commands (npm 5.0.0)

### ISSUE THREAD

**Issue by alaindeurveilher** [npm/npm#16877]
#### I'm opening this issue because:

  - [ ] npm is crashing.
  - [x] npm is producing an incorrect install.
  - [x] npm is doing something I don't understand.
  - [ ] Other (_see below for feature requests_):

#### What's going wrong?
Eratic behaviour of npm 5.0.0: sucessive `npm install` instructions install, then remove, then install packages

#### How can the CLI team reproduce the problem?

1. `ng new angular-cli-model --style=scss --routing --skip-install`
1. `npm install` -> check node_modules/ folder content
1. `npm install` -> check node_modules/ folder content
1. `npm install` -> check node_modules/ folder content

### supporting information:

 - `npm -v` prints: 5.0.0
 - `node -v` prints: v8.0.0
 - `npm config get registry` prints: https://registry.npmjs.org/
 - Windows, OS X/macOS, or Linux?: Windows 7 + nvm 0.24.6 + Git Bash
 - Network issues:
   - Geographic location where npm was run:
   - [x] I use a proxy to connect to the npm registry.
   - [x] I use a proxy to connect to the web.
   - [x] I use a proxy when downloading Git repos.
   - [ ] I access the npm registry via a VPN
   - [ ] I don't use a proxy, but have limited or unreliable internet access.
 - Container:
   - [ ] I develop using Vagrant on Windows.
   - [ ] I develop using Vagrant on OS X or Linux.
   - [ ] I develop / deploy using Docker.
   - [ ] I deploy to a PaaS (Triton, Heroku).

### Additional information

Here is the console commands and outputs:
```bash
exf238@PC130009853 MINGW64 /c/Users/exf238/webserver/toolsuite/_templates (master)
$ cd angular-cli-model/

exf238@PC130009853 MINGW64 /c/Users/exf238/webserver/toolsuite/_templates/angular-cli-model (master)
$ npm install
added 14 packages in 5.774s

exf238@PC130009853 MINGW64 /c/Users/exf238/webserver/toolsuite/_templates/angular-cli-model (master)
$ ls
e2e/  karma.conf.js  node_modules/  package.json  protractor.conf.js  README.md  src/  tsconfig.json  tslint.json

exf238@PC130009853 MINGW64 /c/Users/exf238/webserver/toolsuite/_templates/angular-cli-model (master)
$ npm install
updated 1 package in 1.922s

exf238@PC130009853 MINGW64 /c/Users/exf238/webserver/toolsuite/_templates/angular-cli-model (master)
$ ls
e2e/  karma.conf.js  package.json  protractor.conf.js  README.md  src/  tsconfig.json  tslint.json

exf238@PC130009853 MINGW64 /c/Users/exf238/webserver/toolsuite/_templates/angular-cli-model (master)
$ npm install
added 13 packages and updated 1 package in 11.884s

exf238@PC130009853 MINGW64 /c/Users/exf238/webserver/toolsuite/_templates/angular-cli-model (master)
$ ls
e2e/  karma.conf.js  node_modules/  package.json  protractor.conf.js  README.md  src/  tsconfig.json  tslint.json

exf238@PC130009853 MINGW64 /c/Users/exf238/webserver/toolsuite/_templates/angular-cli-model (master)
$ ls node_modules/
'@angular'/   core-js/   rxjs/   symbol-observable/   zone.js/

exf238@PC130009853 MINGW64 /c/Users/exf238/webserver/toolsuite/_templates/angular-cli-model (master)
$

```

* As we can see, the 1st `npm install` installs only 14 packages.
* Surprisingly, the 2nd `npm install` delete the node_modules folder
* The 3rd `npm install` install again the packages, but only 13 packages, and update 1 (which one??? since they have been removed)

Also, it seems that all the `devDependencies` are simply ignored!

### package.json generated by the cli:
```json
{
  "name": "angular-cli-model",
  "version": "0.0.0",
  "license": "MIT",
  "scripts": {
    "ng": "ng",
    "start": "ng serve",
    "build": "ng build",
    "test": "ng test",
    "lint": "ng lint",
    "e2e": "ng e2e"
  },
  "private": true,
  "dependencies": {
    "@angular/animations": "^4.0.0",
    "@angular/common": "^4.0.0",
    "@angular/compiler": "^4.0.0",
    "@angular/core": "^4.0.0",
    "@angular/forms": "^4.0.0",
    "@angular/http": "^4.0.0",
    "@angular/platform-browser": "^4.0.0",
    "@angular/platform-browser-dynamic": "^4.0.0",
    "@angular/router": "^4.0.0",
    "core-js": "^2.4.1",
    "rxjs": "^5.1.0",
    "zone.js": "^0.8.4"
  },
  "devDependencies": {
    "@angular/cli": "1.1.0",
    "@angular/compiler-cli": "^4.0.0",
    "@angular/language-service": "^4.0.0",
    "@types/jasmine": "2.5.45",
    "@types/node": "~6.0.60",
    "codelyzer": "~3.0.1",
    "jasmine-core": "~2.6.2",
    "jasmine-spec-reporter": "~4.1.0",
    "karma": "~1.7.0",
    "karma-chrome-launcher": "~2.1.1",
    "karma-cli": "~1.0.1",
    "karma-jasmine": "~1.1.0",
    "karma-jasmine-html-reporter": "^0.2.2",
    "karma-coverage-istanbul-reporter": "^1.2.1",
    "protractor": "~5.1.2",
    "ts-node": "~3.0.4",
    "tslint": "~5.3.2",
    "typescript": "~2.3.3"
  }
}
```

**Comment by alaindeurveilher** [comment]
For information, I tried with npm 5.0.1, and the issue is reproduced.

Another thing to notice, I tried the same config on my OSX machine, and I don't experience the issue.

**Comment by alaindeurveilher** [comment]
Potentially a duplicate of #16839 as well.

**Comment by matthewmeyer** [comment]
This issue is kicking my butt too. I have to delete the lock file every time I do an install, or run with --no-shrinkwrap, otherwise I get an odd subset of packages.

**Comment by alaindeurveilher** [comment]
I actually first experienced the issue, when executing yarn after updating from node 7.10.0 to 8.0.0+npm5. I did not know what was wrong till the moment I executed "npm install" to compare the results with yarn.

Unfortunately I had to switch back to node 7.10.0+npm4 to go on on my project, which is sad because it means less feedback for node8+npm5. But I cannot be blocked on my dev till this issue is fixed.
